src/keycloak-theme/login/template.component.ts

```diff
@@ -25,73 +14,29 @@ import { I18nService } from "./common/services/i18n.service";
   selector: "kc-login-template",
   templateUrl: "./template.component.html",
   standalone: true,
-  imports: [
-    RouterLink,
-    RouterLinkActive,
-    CommonModule,
-    RouterOutlet,
-    KcClassPipe,
-    SanitizeHtmlPipe,
-  ],
-  providers: [KcClassPipe, I18nService, CommonModule],
+  imports: [RouterLink, RouterLinkActive, AsyncPipe, KcClassPipe, SanitizeHtmlPipe],
+  providers: [KcClassPipe]
 })
-export class TemplateComponent implements OnInit {
+export class TemplateComponent implements OnInit, OnDestroy {
   stylesheetsLoaded?: boolean = false;
-  kcContext: KcContext = window.kcContext;
+  kcContext: KcContext = inject(KC_CONTEXT);
+  i18nService = inject(I18nService);
+  private kcClassPipe = inject(KcClassPipe);
+  private dynamicStyleLoader = inject(DynamicStyleLoader);
   displayRequiredFields = false;
-  i18n$?: Observable<GenericI18n_noJsx<any> | null>;
-  i18n?: GenericI18n_noJsx<any> | null;
 
-  // Child data
-  displayInfo?: boolean = true;
-  headerNode?: TemplateRef<any>;
-  infoNode?: TemplateRef<any>;
-  socialProvidersNode?: TemplateRef<any>;
+  displayInfo = input(true);
+  displayMessage = input(true);
 
-  constructor(
-    private cdref: ChangeDetectorRef,
-    public i18nService: I18nService,
-    private kcClassPipe: KcClassPipe,
-    private dynamicStyleLoader: DynamicStyleLoader,
-    private router: Router
-  ) {}
+  subscription: Unsubscribable | undefined;
 
   ngOnInit() {
     this.loadStyles();
-    this.setNavigationToPageId();
     this.applyKcIndexClasses();
-    this.i18nService.i18n$.subscribe((i18n) => {
-      this.i18n = i18n;
-    });
-  }
-
-  onRouterOutletActivate(event: any) {
-    this.cdref.detectChanges();
-    // Here you can set Nodes from children like headerNode
-    this.headerNode = event.headerNode;
-    this.infoNode = event.infoNode;
-    this.socialProvidersNode = event.socialProvidersNode;
-    this.displayInfo = event.displayInfo;
-  }
-
-  private setNavigationToPageId() {
-    this.router.navigate(
-      [
-        "login",
-        { outlets: { login: this.trimPageIdSuffix(window.kcContext.pageId) } },
-      ],
-      { skipLocationChange: true }
-    );
-  }
-
-  private trimPageIdSuffix(pageId: string): string {
-    if (pageId.length > 4) {
-      return pageId.substring(0, pageId.length - 4);
-    }
-    return pageId;
   }
 
   private applyKcIndexClasses() {
+    // TODO: maybe is better to load classes in app.component ?
```

Comment:
are those classes provided in account as well? I haven't done any work on the account side.